library/src/main/java/com/novoda/downloadmanager/NotificationChannelCreator.java

```diff
@@ -1,11 +1,10 @@
 package com.novoda.downloadmanager;
 
 import android.app.NotificationChannel;
+import android.os.Build;
+import android.support.annotation.RequiresApi;
 
 interface NotificationChannelCreator {
-
-    Optional<NotificationChannel> createNotificationChannel();
-
-    String getNotificationChannelId();
-
+    @RequiresApi(Build.VERSION_CODES.O)
```

Comment:
Can this annotation live on the interface rather than the method? Just thinking that it applies to everything in here atm. Not sure if that is just being a nit 😄

Reply:
We actually need it in both places - the callers of this method only know about the interface so they need it on the interface method. Unfortunately method annotations aren't inherited from interfaces so unless we have it on the actual implementation as well, we would need the runtime `Build.VERSION_CODES` check.